bears/general/AnnotationBear.py

```diff
@@ -0,0 +1,172 @@
+        if match_pos:
+            m_comm = list(get_range(file, filename, match_pos, multi_comm))
+            s_comm = list(get_range(file, filename, match_pos, single_comm))
+            m_str = list(get_range(file, filename, match_pos, multi_str))
+            s_str = list(get_range(file, filename, match_pos, single_str))
+        yield HiddenResult(self, m_comm)
+        yield HiddenResult(self, s_comm)
+        yield HiddenResult(self, m_str)
+        yield HiddenResult(self, s_str)
+
+
+def get_range(file, filename, match_pos, string_comment):
+    """
+    Checks if range is valid and then yields it.
+    """
+    search_range = []
+    for i in string_comment:
+        if i in match_pos:
+            search_range.append(i)
+    search_range = [(calc_line_col(file, start), calc_line_col(file, end))
+                    for (start, end) in search_range]
+
+    for i in search_range:
+        yield SourceRange.from_values(filename,
+                                      start_line=i[0][0],
+                                      start_column=i[0][1],
+                                      end_line=i[1][0],
+                                      end_column=i[1][1])
+
+
+def find_string_comment(file, annot, escape):
+    """
+    gives all instances of strings or multiline comments found within
+    the file, even if they are nested in other strings or comments
+    """
+    if escape:
+        search_func = unescaped_search_in_between
+    else:
+        search_func = search_in_between
+    found_pos = ()
+    for annot_type in annot:
+        found_pos += tuple(search_func(annot_type, annot[annot_type], file))
+    if found_pos:
+        found_pos = tuple((i.begin.range[0], i.end.range[1])
+                          for i in found_pos)
+    return found_pos
+
+
+def find_singleline_comments(file, annot, match_pos):
+    """
+    Finds all single-line comments outside of other string or comments.
+    """
+    single_comm = []
+    for comment_type in annot:
+        for i in search_for(comment_type, file):
+            if not in_range_list(match_pos, (i.start(), i.end())):
+                end = file.find('\n', i.start())
+                if end != -1:
+                    single_comm.append((i.start(), end))
+                else:
+                    end = len(file) + 1
+                    single_comm.append((i.start(), end))
+    return tuple(single_comm)
+
+
+def in_range_list(outside_range_list, inside_range):
+    """
+    finds if a given 'range' is inside a any of the 'ranges' in
+    a list of ranges.
+    """
+    for outside_range in outside_range_list:
+        if inside_range == outside_range:
+            continue
+
+        elif inside_range[0] == outside_range[0]:
+            if inside_range[1] < outside_range[1]:
+                return True
+
+        elif inside_range[0] in range(outside_range[0], outside_range[1]):
+            return True
+    return False
+
+
+def calc_line_col(file, pos_to_find):
```

Comment:
This function seems super useful. I remember @sudheesh001 struggling with this in `ESLintBear`.
Can we possible move it into SourceRange or something ? so that bear users can use it ?

I think someone else also needed it for a bear recently.

Currently they seem to be doing "".join(file) but this seems to be better in terms of performance